pkg/networkserver/mac/adr.go

```diff
@@ -368,10 +369,64 @@ func clampTxPowerRange(
 	}
 }
 
-func clampNbTrans(dev *ttnpb.EndDevice, defaults *ttnpb.MACSettings, nbTrans uint32) uint32 {
+// DataRateIndexOverridesOf returns the per-data rate index overrides of the given dynamic ADR settings.
+func DataRateIndexOverridesOf(
+	overrides *ttnpb.ADRSettings_DynamicMode_Overrides, drIdx ttnpb.DataRateIndex,
+) *ttnpb.ADRSettings_DynamicMode_PerDataRateIndexOverride {
+	switch drIdx {
```

Comment:
This is the only 'ugly' part of modelling the overrides as fields. But, we have to do this only once in the whole codebase.